### ISSUE TITLE
prefer-self-closing-tags false positive when put component inside ng-content tag

### ISSUE THREAD

**Issue by 5im0n** [angular-eslint/angular-eslint#1970]
```JSON
{
  "rules": {
    "@angular-eslint/template/prefer-self-closing-tags": "error",
  }
}
```

✅ Valid Code
```HTML
<ng-content select="[emptySlate]">
   Empty slate
</ng-content>
```

❌ Invalid Code
```HTML
<ng-content select="[emptySlate]">
   <h2>Empty slate</h2>
</ng-content>
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `18.2.0` |
| `@angular-eslint/template-parser`         | `18.2.0` |
| `@typescript-eslint/parser`               | `7.18.0` |
| `ESLint`                                  | `8.57.0` |
| `node`                                    | `20.16.0` |

```sh
@angular-devkit/architect    0.1801.3
@angular-devkit/core         18.1.3
@angular-devkit/schematics   18.1.3
@schematics/angular          18.1.3
rxjs                         7.8.1
typescript                   5.5.4
zone.js                      0.14.10

```

- [X ] I have tried restarting my IDE and the issue persists.
- [ X] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.
